src/algebra/group/pi.lean

```diff
@@ -147,6 +147,26 @@ end mul_hom
 
 section mul_hom
 
+/-- A family of monoid homomorphisms `f a : γ →* β a` defines a monoid homomorphism
+`pi.monoid_mul_hom f : γ →* Π a, β a` given by `pi.monoid_mul_hom f x b = f b x`. -/
+@[to_additive "A family of additive monoid homomorphisms `f a : γ →+ β a` defines a monoid
+homomorphism `pi.monoid_add_hom f : γ →+ Π a, β a` given by `pi.monoid_add_hom f x b
+= f b x`.", simps]
+def pi.monoid_mul_hom {γ : Type w} [Π i, monoid (f i)] [monoid γ]
+  (g : Π i, γ →* f i) : γ →* Π i, f i :=
+{ to_fun := λ x b, g b x,
+  map_one' := funext $ λ z, (g z).map_one,
+  map_mul' := λ x y, funext $ λ z, (g z).map_mul x y, }
+
+/-- A family of mul_hom `f a : γ →ₙ* β a` defines a mul_hom `pi.mul_hom f : γ →ₙ* Π a, β a`
+given by `pi.mul_hom f x b = f b x`. -/
+@[to_additive "A family of add_hom `f a : γ → β a` defines a add_hom `pi.add_hom
+f : γ → Π a, β a` given by `pi.add_hom f x b = f b x`.", simps]
+def pi.mul_hom {γ : Type w} [Π i, has_mul (f i)] [has_mul γ]
+  (g : Π i, γ →ₙ* f i) : γ →ₙ* Π i, f i :=
+{ to_fun := λ x b, g b x,
+  map_mul' := λ x y, funext $ λ z, (g z).map_mul x y, }
```

Comment:
This one should come before `pi.monoid_hom`; then you can define that in terms of this.

Reply:
I made the changes so that definitions are reused. The `..` mechanism is really useful